receiver/awscontainerinsightreceiver/internal/neuron/neuron_monitor_scraper.go

```diff
@@ -0,0 +1,208 @@
+	}
+	if opts.HostInfoProvider == nil {
+		return nil, errors.New("cluster name provider cannot be nil")
+	}
+
+	scrapeConfig := &config.ScrapeConfig{
+		ScrapeInterval: model.Duration(collectionInterval),
+		ScrapeTimeout:  model.Duration(collectionInterval),
+		JobName:        jobName,
+		Scheme:         "http",
+		MetricsPath:    "/metrics",
+		ServiceDiscoveryConfigs: discovery.Configs{
+			&kubernetes.SDConfig{
+				Role: kubernetes.RoleService,
+				NamespaceDiscovery: kubernetes.NamespaceDiscovery{
+					IncludeOwnNamespace: true,
+				},
+				Selectors: []kubernetes.SelectorConfig{
+					{
+						Role:  kubernetes.RoleService,
+						Label: "k8s-app=neuron-monitor-service",
+					},
+				},
+				AttachMetadata: kubernetes.AttachMetadataConfig{
+					Node: true,
+				},
+			},
+		},
+		RelabelConfigs: []*relabel.Config{
+			{
+				SourceLabels: model.LabelNames{"__address__"},
+				Regex:        relabel.MustNewRegexp("([^:]+)(?::\\d+)?"),
+				Replacement:  "${1}:8000",
+				TargetLabel:  "__address__",
+				Action:       relabel.Replace,
+			},
+		},
+		MetricRelabelConfigs: []*relabel.Config{
+			{
+				SourceLabels: model.LabelNames{"__name__"},
+				Regex:        relabel.MustNewRegexp("neuron.*|system_.*|execution_.*"),
+				Action:       relabel.Keep,
+			},
+			{
+				SourceLabels: model.LabelNames{"instance_name"},
+				TargetLabel:  "NodeName",
+				Regex:        relabel.MustNewRegexp("(.*)"),
+				Replacement:  "${1}",
+				Action:       relabel.Replace,
+			},
+			{
+				SourceLabels: model.LabelNames{"instance_id"},
+				TargetLabel:  "InstanceId",
+				Regex:        relabel.MustNewRegexp("(.*)"),
+				Replacement:  "${1}",
+				Action:       relabel.Replace,
+			},
+			{
+				SourceLabels: model.LabelNames{"neuroncore"},
+				TargetLabel:  "DeviceId",
+				Regex:        relabel.MustNewRegexp("(.*)"),
+				Replacement:  "${1}",
+				Action:       relabel.Replace,
+			},
+			// hacky way to inject static values (clusterName) to label set without additional processor
+			// relabel looks up an existing label then creates another label with given key (TargetLabel) and value (static)
+			{
+				SourceLabels: model.LabelNames{"instance_id"},
+				TargetLabel:  "ClusterName",
+				Regex:        relabel.MustNewRegexp("neuron.*"),
+				Replacement:  opts.HostInfoProvider.GetClusterName(),
+				Action:       relabel.Replace,
+			},
+			// hacky way to inject static values (namespace, podName, containerName) to label set without additional processor,
+			// we need these labels only when we have neuronCore label specified in logs
+			// we process and replace with actual podName, namespace and ContainerName
+			// relabel looks up an existing label then creates another label with given key (TargetLabel) and value (static)
+			{
+				SourceLabels: model.LabelNames{"neuroncore"},
+				TargetLabel:  "Namespace",
+				Regex:        relabel.MustNewRegexp("(.*)"),
+				Replacement:  "${1}",
+				Action:       relabel.Replace,
+			},
```

Comment:
Well this makes it easier in Agent to replace the value by PodName, Namespace by passing in the value. Otherwise we would need to add new labels.

Only certain metrics will have these labels so if we attach these labels here it would be easier to replace them.

Reply:
We discussed offline.  The idea now is that if we put the labels on here, then the agent can just replace them where it finds them, as opposed to having to know which metrics should get the PodName/ContainerName/etc. labels.  But if possible, I like to avoid these kinds of implicit contracts between components - it's fragile and very easy to forget why we put the labels on here with some placeholder value.  So we will see how it looks when we get to the agent work.